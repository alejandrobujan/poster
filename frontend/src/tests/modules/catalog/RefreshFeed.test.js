import RefreshFeed from "../../../modules/catalog/components/RefreshFeed";
import configureMockStore from 'redux-mock-store';
import renderer from "react-test-renderer";
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import { refreshFeedMock } from "../../state/RefreshFeed.mock";
import { render, fireEvent } from '@testing-library/react';

describe("RefreshFeed", () => {
	
	const middlewares = [thunk];
	const mockStore = configureMockStore(middlewares);
	let store;
	store = mockStore(refreshFeedMock);
	
	it("renders correctly", () => {
		const tree = renderer
			.create(
				<Provider store={store}>
					<RefreshFeed />
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	describe('refreshPage', () => {
		const { reload } = window.location;
	
		beforeAll(() => {
		  Object.defineProperty(window, 'location', {
			writable: true,
			value: { reload: jest.fn() },
		  });
		});
	
		afterAll(() => {
		  window.location.reload = reload;
		});
	
		it('reloads the window', () => {
			const { getByRole } = render(<RefreshFeed />);
    
			fireEvent.click(getByRole('button'));
		  expect(window.location.reload).toHaveBeenCalled();
		});
	});
});
