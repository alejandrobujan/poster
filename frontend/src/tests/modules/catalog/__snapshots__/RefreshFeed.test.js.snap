// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefreshFeed renders correctly 1`] = `
<div
  className="m-2 alert alert-warning d-flex justify-content-between align-items-center"
  role="alert"
>
  <span
    className="mb-0"
  >
    New posts were added. Refresh the feed to get the latest updates!
  </span>
  <span
    className="text-right"
  >
    <button
      className="px-2 btn btn-secondary d-flex justify-content-between align-items-center"
      onClick={[Function]}
      type="button"
    >
      <svg
        className="bi bi-arrow-repeat"
        fill="currentColor"
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11.534 7h3.932a.25.25 0 0 1 .192.41l-1.966 2.36a.25.25 0 0 1-.384 0l-1.966-2.36a.25.25 0 0 1 .192-.41zm-11 2h3.932a.25.25 0 0 0 .192-.41L2.692 6.23a.25.25 0 0 0-.384 0L.342 8.59A.25.25 0 0 0 .534 9z"
        />
        <path
          d="M8 3c-1.552 0-2.94.707-3.857 1.818a.5.5 0 1 1-.771-.636A6.002 6.002 0 0 1 13.917 7H12.9A5.002 5.002 0 0 0 8 3M3.1 9a5.002 5.002 0 0 0 8.757 2.182.5.5 0 1 1 .771.636A6.002 6.002 0 0 1 2.083 9z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </span>
</div>
`;
